src/colmap/estimators/alignment.cc

```diff
@@ -335,15 +339,81 @@ std::vector<ImageAlignmentError> ComputeImageAlignmentError(
   return errors;
 }
 
+bool AlignReconstructionsViaPoints(const Reconstruction& src_reconstruction,
+                                   const Reconstruction& tgt_reconstruction,
+                                   const size_t min_common_observations,
+                                   const double max_error,
+                                   const double min_inlier_ratio,
+                                   Sim3d* tgt_from_src) {
+  CHECK_GT(min_common_observations, 0);
+  CHECK_GT(max_error, 0.0);
+  CHECK_GE(min_inlier_ratio, 0.0);
+  CHECK_LE(min_inlier_ratio, 1.0);
+
+  std::vector<Eigen::Vector3d> xyz_src;
+  std::vector<Eigen::Vector3d> xyz_tgt;
+  // Associate 3D points using point2D_idx
+  for (auto& p3D_p : src_reconstruction.Points3D()) {
+    // Count how often a 3D point in tgt is associated to this 3D point
+    std::unordered_map<point3D_t, size_t> counts;
```

Comment:
nit: could be defined outside the loop and then clear() here to reduce malloc's.